auth-client-springsecurity/src/main/java/gov/usgs/cida/auth/client/springsecurity/CidaAuthenticationProvider.java

```
@@ -0,0 +1,82 @@
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.InternalAuthenticationServiceException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is a minimal implementation of AuthenticationProvider
+ * This just delegates to a CIDA IAuthClient implementation for authentication
+ * and provisioning.  The IAuthClient is wired in by the Spring configuration.
+ * 
+ * Note:  See the src/sample directory for an example of how to use this within
+ * a Spring app.
+ * 
+ * @author eeverman
+ */
+@Component
+public class CidaAuthenticationProvider implements AuthenticationProvider {
+	private static final Logger LOG = LoggerFactory.getLogger(CidaAuthenticationProvider.class);
+	
+	protected IAuthClient authClient;
+	
+	@Autowired
+	public CidaAuthenticationProvider(final IAuthClient authClient) {
+		this.authClient = authClient;
+	}
+	
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        String name = authentication.getName();
+        String password = authentication.getCredentials().toString();
+		
+		AuthToken token = null;
+		
+		try {
+			token = authClient.getNewToken(name, password);
+		} catch (LoginException ex) {
+			throw new BadCredentialsException("Invalid username or password");
+		} catch (WebApplicationException ex) {
+			LOG.error("Unable to process authentication request", ex);
+			throw new InternalAuthenticationServiceException(
```

Yeah, I thought about passing in the root exception.  I decided to swallow the exception because I didn’t want it passed up to the UI layer, where it might give away secret info (like the url of our Auth server along w/ the credentials it was sending).

Since I’m logging the issue here, I couldn’t think of anything more I could do w/ it farther up the chain.  I could try to pull more detail out of the exception for logging.

I believe Spring will, by default, ‘Do the right thing’ as long as I wrap it in a WebApplicationException.  Would people prefer that to what I’m doing here?